src/components/ebay-date-textbox/component.js

```diff
@@ -0,0 +1,136 @@
+// @ts-check
+
+import Expander from 'makeup-expander';
+import { toISO } from '../ebay-calendar/component';
+import * as eventUtils from '../../common/event-utils';
+
+const MIN_WIDTH_FOR_DOUBLE_PANE = 600;
+
+/**
+ * @typedef {import('../ebay-calendar/component').DayISO} DayISO
+ * @typedef {{
+ *   locale?: string,
+ *   range?: boolean,
+ *   disableBefore?: Date | number | string,
+ *   disableAfter?: Date | number | string,
+ *   disableWeekdays?: number[],
+ *   disableList?: (Date | number | string)[],
+ *   getA11yShowMonthText?: (monthName: string) => string,
+ *   a11yRangeStartText?: string,
+ *   a11yInRangeText?: string,
+ *   a11yRangeEndText?: string,
+ * }} Input
+ * @typedef {{
+ *   numMonths: number,
+ *   firstSelected: DayISO | undefined,
+ *   secondSelected: DayISO | undefined,
+ *   popover: boolean,
+ * }} State
+ * @extends {Marko.Component<Input, State>}
+ */
+export default class extends Marko.Component {
+    onCreate() {
+        this.state = {
+            numMonths: 1,
+            firstSelected: undefined,
+            secondSelected: undefined,
+            popover: false,
+        };
+
+        this.calculateNumMonths();
+    }
+
+    onMount() {
+        this.expander = new Expander(/** @type {HTMLElement} */ (this.el), {
+            hostSelector: '.ebay-date-textbox--main > .icon-btn',
+            contentSelector: '.date-textbox__popover',
+            expandOnClick: true,
+            autoCollapse: true,
+        });
+        this.subscribeTo(eventUtils.resizeUtil).on('resize', this.calculateNumMonths.bind(this));
```

Comment:
There might be a better way to do this

Maybe we can use subscribe tag (see video)
In that case we only attach the event when the pane is opened. We shouldn't be recalculating when the pane is closed.